inspirehep/ext/jinja_filters/record.py

```diff
@@ -458,3 +427,105 @@ def is_upper(s):
     def split_author_name(name):
         new_name = name.split(',')
         return ' '.join(reversed(new_name))
+
+    @app.template_filter()
+    def publication_info(record):
+        """Displays inline publication and conference information"""
+        result = {}
+        out = []
+        """Publication info line"""
+        if 'publication_info' in record:
+            journal_title, journal_volume, year, journal_issue, pages = \
+                ('', '', '', '', '')
+            for pub_info in record['publication_info']:
+                if 'journal_title' in pub_info:
+                    journal_title = '<i>' + pub_info['journal_title'] + '</i>'
+                    if 'journal_volume' in pub_info:
+                        journal_volume = ' ' + pub_info['journal_volume']
+                    if 'year' in pub_info:
+                        year = ' (' + str(pub_info['year']) + ')'
+                    if 'journal_issue' in pub_info:
+                        journal_issue = ' ' + pub_info['journal_issue'] + ', '
+                    if 'page_artid' in pub_info:
+                        pages = ' ' + pub_info['page_artid']
+                    out.append(journal_title + journal_volume +
+                               year + journal_issue + pages)
+            if out:
+                result['pub_info'] = out
+            if not result:
+                for field in record['publication_info']:
+                    if 'pubinfo_freetext' in field:
+                        out.append(field['pubinfo_freetext'])
+                        result['pub_info'] = out
+                        break
+            """Conference info line"""
+            for element in record['publication_info']:
+                if 'conference_recid' in element \
+                        and 'parent_recid' in element:
+                    conference_rec = es.get_source(
+                        index='conferences',
+                        id=element['conference_recid'],
+                        doc_type='record')
+                    ctx = {
+                        "parent_recid": str(
+                            element['parent_recid']),
+                        "conference_recid": str(
+                            element['conference_recid']),
+                        "conference_title": conference_rec['title']
+                    }
+                    if result:
+                        result['conf_info'] = render_macro_from_template(
+                            name="conf_with_title",
+                            template="format/record/Conference_info_macros.tpl",
+                            ctx=ctx)
+                        break
+                    else:
+                        if 'page_artid' in element:
+                            ctx.update(
```

Comment:
if `page_artid` is there you can update the `ctx`. Then you can remove the `else` statement and the duplicated code:

```python
result['conf_info'] = render_macro_from_template(
 +                                name="conf_without_title",
 +                                template="format/record/Conference_info_macros.tpl",
 +                                ctx=ctx)
 +                            break
```